src/error.rs

```diff
@@ -624,6 +625,10 @@ pub enum ErrorKind {
     /// A custom value produced by user code.
     #[error("Custom user error")]
     Custom(Arc<dyn Any + Send + Sync>),
+
+    /// A method was called on a client that was shut down.
+    #[error("Client has been shut down")]
+    Shutdown,
```

Comment:
I don't see this error kind constructed anywhere; should we be returning this somewhere in `execute_operation`?

Reply:
Whoops, yup, forgot to follow through here.  Fixed.